src/connector/src/parser/upsert_parser.rs

```diff
@@ -50,21 +52,38 @@ async fn build_accessor_builder(
     }
 }
 
+fn check_rw_default_key(columns: &Vec<SourceColumnDesc>) -> bool {
+    for col in columns {
+        if col.name.starts_with(DEFAULT_KEY_COLUMN_NAME) {
+            return true;
+        }
+    }
+    false
+}
+
 impl UpsertParser {
     pub async fn new(
         props: ParserProperties,
         rw_columns: Vec<SourceColumnDesc>,
         source_ctx: SourceContextRef,
     ) -> Result<Self> {
-        // TODO: should check where to find name(key or value) when there is such circumstances.
-        let avro_primary_key_column_name =
+        let mut avro_primary_key_column_name = None;
+        let key_builder: AccessBuilderImpl;
+        // check whether columns has `DEFAULT_KEY_COLUMN_NAME`, if so, the key accessor should be
+        // bytes
+        if check_rw_default_key(&rw_columns) {
+            key_builder = AccessBuilderImpl::Bytes(BytesAccessBuilder::new(
+                EncodingProperties::Bytes(BytesProperties {
+                    column_name: Some(DEFAULT_KEY_COLUMN_NAME.into()),
+                }),
+            )?);
```

Comment:
a little curious here, I think if there is a key schema in schema registry, the column name should be `_rw_key`, too. Making the default encode to `bytes` should be done when inferring the table schema.